### PR TITLE
add new package manager prefs section

### PR DESCRIPTION
**Pull request by mjkkirschner**
### Purpose

- adds a new tab for package manager client settings.
- adds expanders with empty labels to be filled in.
- increases width of tabs by 50 pixels.
- ~~increases width of window by 50pixels.~~
- renames some methods to follow c# naming standards.
- fixes a bug with the labels for geometry scaling-  both medium and large had the same description.

- [x] ~~I would like to make the window resizable horizontally to deal with localized text being longer - but need to check with Morpheus team if that is okay.~~ We're going to wait on this.

Screen Shot 2021-06-04 at 6 35 16 PM

### Declarations

Check these if you believe they are true

- [x] The codebase is in a better state after this PR
- [x] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [x] The level of testing this PR includes is appropriate
- [x] User facing strings, if any, are extracted into `*.resx` files
- [ ] All tests pass using the self-service CI.
- [x] Snapshot of UI changes, if any.
- [ ] Changes to the API follow [Semantic Versioning](https://github.com/DynamoDS/Dynamo/wiki/Dynamo-Versions) and are documented in the [API Changes](https://github.com/DynamoDS/Dynamo/wiki/API-Changes) document.
- [ ] This PR modifies some build requirements and the readme is updated

### CODE REVIEW

**Comment by QilongTang** [comment]
@Jingyi-Wen Can you take a look? The new sub panel tabs looks a bit strange to me, and I wonder if we can make the tabs wider and wordwrap instead of having to make the whole dialog wider? @RobertGlobant20 What do you think?

**Review by aparajit-pratap — APPROVED** [review]

**Comment by Jingyi-Wen** [comment]
> @Jingyi-Wen Can you take a look? The new sub panel tabs looks a bit strange to me, and I wonder if we can make the tabs wider and wordwrap instead of having to make the whole dialog wider? @RobertGlobant20 What do you think?

Looking at other tabs we already have, I think we can make the tabs wider for all of them and leave the dialog width as it is

**Comment by mjkkirschner** [comment]
@Jingyi-Wen if we don't make the dialog wider, then some content wraps in strange ways, let me get you a screenshot.

**Comment by mjkkirschner** [comment]
Screen Shot 2021-06-07 at 12 16 14 PM

I'm remembering wrong- 
@QilongTang @Jingyi-Wen - here is the dialog with wider tabs, but at 530pixels - it seems something is wrong with the word wrap for the geometry scaling description. It's wrapping, but also looks cutoff.

**Comment by QilongTang** [comment]
> Screen Shot 2021-06-07 at 12 16 14 PM
> 
> I'm remembering wrong-
> @QilongTang @Jingyi-Wen - here is the dialog with wider tabs, but at 530pixels - it seems something is wrong with the word wrap for the geometry scaling description. It's wrapping, but also looks cutoff. Maybe it just needs a larger margin.
> 
> Do you have a preference @Jingyi-Wen? The faster I get this merged, the faster we can move on to other UI for the package manager prefs.

 @mjkkirschner Maybe we wrap the localization problem into a different task and PR? I feel the impact is larger than we thought?

File: src/DynamoCoreWpf/Views/Menu/PreferencesView.xaml.cs
Comment (QilongTang) [inline src/DynamoCoreWpf/Views/Menu/PreferencesView.xaml.cs:21]: Thanks! not used?

Comment (mjkkirschner) [reply src/DynamoCoreWpf/Views/Menu/PreferencesView.xaml.cs:21]: yep

File: src/DynamoCoreWpf/Properties/Resources.resx
Comment (QilongTang) [inline src/DynamoCoreWpf/Properties/Resources.resx:2403]: Can you look for same string within Dynamo code base? I think we already have this string.
image

File: src/DynamoCoreWpf/Properties/Resources.resx
Comment (QilongTang) [inline src/DynamoCoreWpf/Properties/Resources.resx:2406]: Do you want to reuse the string `Manage Node and Package Path`? Just curious
image

Comment (mjkkirschner) [reply src/DynamoCoreWpf/Properties/Resources.resx:2406]: I was following the design.

File: src/DynamoCoreWpf/Properties/Resources.resx
Comment (mjkkirschner) [reply src/DynamoCoreWpf/Properties/Resources.resx:2403]: sure, will look.

File: src/DynamoCoreWpf/Properties/Resources.resx
Comment (QilongTang) [reply src/DynamoCoreWpf/Properties/Resources.resx:2406]: @mjkkirschner I see. Your call between you and @Jingyi-Wen

Comment (mjkkirschner) [reply src/DynamoCoreWpf/Properties/Resources.resx:2406]: @Jingyi-Wen thoughts?

File: src/DynamoCoreWpf/Properties/Resources.resx
Comment (QilongTang) [inline src/DynamoCoreWpf/Properties/Resources.resx:2409]: I did not find this one in current UI so assume safe to add :)

**Review by QilongTang — APPROVED** [review]
Looks promising. @mjkkirschner Once comments addressed, LGTM

File: src/DynamoCoreWpf/Properties/Resources.resx
Comment (Jingyi-Wen) [reply src/DynamoCoreWpf/Properties/Resources.resx:2406]: > @Jingyi-Wen thoughts on changing `package paths` to `node and package paths` or `manage node and package paths`?

Seems like `node and package paths` would be more accurate. (No need to add `manage` since we are not using verbs in other expanders or items.) I'll update the Figma

Comment (mjkkirschner) [reply src/DynamoCoreWpf/Properties/Resources.resx:2406]: 👍

File: src/DynamoCoreWpf/Views/Menu/PreferencesView.xaml.cs
Comment (mjkkirschner) [inline src/DynamoCoreWpf/Views/Menu/PreferencesView.xaml.cs:64]: geometry scaling descriptions were repeated for medium and large, and extra large was wrong.

**Comment by RobertGlobant20** [comment]
> @Jingyi-Wen Can you take a look? The new sub panel tabs looks a bit strange to me, and I wonder if we can make the tabs wider and wordwrap instead of having to make the whole dialog wider? @RobertGlobant20 What do you think?

@QilongTang I'm not sure about the consequences of making the tabs wider and wordwrap (without making the dialog wider). but as I was guessing if the tabs (and content) width changes we will face several alignment problems like the text in the Geometry Scaling expander.

**Comment by mjkkirschner** [comment]
I'm going to merge this after one more test run.

**Review by RobertGlobant20 — APPROVED** [review]
LGTM, @mjkkirschner could you include a screenshot of the Features and  Visual Settingstab after changing the TabPanel.Width

**Comment by mjkkirschner** [comment]
@RobertGlobant20 
Screen Shot 2021-06-07 at 2 12 44 PM
Screen Shot 2021-06-07 at 2 12 32 PM

**Pull request merged**
